compiler/rustc_codegen_llvm/src/builder/gpu_offload.rs

```diff
@@ -69,6 +71,63 @@ impl<'ll> OffloadGlobals<'ll> {
     }
 }
 
+pub(crate) struct OffloadKernelDims<'ll> {
+    num_workgroups: &'ll Value,
+    threads_per_block: &'ll Value,
+    workgroup_dims: &'ll Value,
+    thread_dims: &'ll Value,
+}
+
+impl<'ll> OffloadKernelDims<'ll> {
+    pub(crate) fn from_operands<'tcx>(
+        builder: &mut Builder<'_, 'll, 'tcx>,
+        workgroup_op: &OperandRef<'tcx, &'ll llvm::Value>,
+        thread_op: &OperandRef<'tcx, &'ll llvm::Value>,
+    ) -> Self {
+        let cx = builder.cx;
+
+        let workgroup_val = match workgroup_op.val {
+            OperandValue::Ref(place) => builder.load(
+                cx.type_array(cx.type_i32(), 3),
```

Comment:
You left this (and the one below) as i32, is that intended?

Reply:
llvm doesn't distinguish those. It just has different ops